doc/.ChangeLog_template

```diff
@@ -48,45 +52,42 @@ Changes made to namelist defaults (e.g., changed parameter values):
 
 Changes to the datasets (e.g., parameter, surface or initial files):
 
-Substantial timing or memory changes: [For timing changes, can check PFS test(s) in the test suite]
+Substantial timing or memory changes:
+[For timing changes, you should at least check the PFS test in the test
+ suite, whether or not you expect timing changes: ensure that its run
+ time has not increased significantly relative to the baseline.]
```

Comment:
@ekluzek @slevisconsulting @negin513 - I've gotten in the habit of checking the run time of the PFS test compared with baselines for nearly every tag (other than those where the changes are very minor). I think that's a good thing for us to do in general; do you agree? If so, is this a good enough way to call this out? I thought about adding this to the testing section, but it doesn't seem to completely fit there, since everything else in the testing section is more an objective pass/fail, whereas this is more subjective. I'm open to suggestions here.

By the way, once we update to a cime version that pulls in https://github.com/ESMCI/cime/pull/3786, this will become easier to check: you'll be able to grep for TPUTCOMP in the TestStatus.log file for the PFS test. At that point, I'll probably update the text in brackets to point that out.

Reply:
This sounds good. I don't consistently check partially because in the past it wasn't reliable. But, it's good to know you think it's reliable now. And if so checking every time significant changes are made makes sense. The test is subjective, but it does result in a PASS/FAIL based on some arbitrary threshold. So I think reporting the result when it fails is what we should do. Machine variability makes me suspect of results that might show a small timing change that might not really be accurate.

Is there just a single PFS test? Machine variability means it might be better to have several unless the PFS test itself does multiple runs, which I don't think it does.

Reply:
> This sounds good. I don't consistently check partially because in the past it wasn't reliable. But, it's good to know you think it's reliable now. And if so checking every time significant changes are made makes sense. The test is subjective, but it does result in a PASS/FAIL based on some arbitrary threshold. So I think reporting the result when it fails is what we should do. Machine variability makes me suspect of results that might show a small timing change that might not really be accurate.
> 
> Is there just a single PFS test? Machine variability means it might be better to have several unless the PFS test itself does multiple runs, which I don't think it does.

I'll be honest: I've been struggling with this one. On the one hand, it would be good for us to notice when changes unintentionally slow the model down significantly. But on the other hand, I want to keep the check of this quick (e.g., no more than a minute or two of our time per tag) and relatively low in computational expense (because our running of the aux_clm test suite already takes so much of the SEWG computing allocation).

To be honest, I'm not sure how reliable it is. Up until now, I've just been checking the time of the test RUN phase, which tends to be reasonably consistent, but still sometimes has a big blip (which occasionally leads me to rerun the test). It does seem like the performance of the PFS test is more stable than many of our other tests, because there is relatively more computation time rather than i/o time, and I think there is significantly less machine variability in computation time compared with i/o time. (The PFS test doesn't do any output, but of course still does all the normal file input.)

I'm hesitant to add multiple PFS tests for the following reasons:

1. They are relatively expensive tests (20-day 1-degree tests currently)

2. Interpreting the results could be time consuming if we have multiple tests that we need to look at

3. I'm not sure how much value this would add if the multiple tests run in succession or all at once: my hunch is that the biggest variability comes from glade slowness, and my hunch (unconfirmed) is that, if one test is experiencing issues, then other tests run at that moment or a close time would be more likely to experience the same issues.

So my feeling has been that a single, high-resolution, long-ish test (as we currently have) does as good a job at averaging out the machine variability as we'd get from multiple tests run all at once.

Regarding the idea of using the PASS/FAIL: The default tolerance is set at 25%, which seems too high. I just saw that there's a way to specify a per-test throughput tolerance, so we could lower the tolerance for the PFS test. But with the way I've set up our standard cs.status script to ignore throughput tolerance failures in general (because there is a lot of variability for many other tests), this wouldn't show up in the cs.status results. So you would need to grep for this line in the PFS's TestStatus file; I'm not sure that has any advantage over grepping for it in TestStatus.log, which is where I put it in ESMCI/CIME#3786? (In principle, I guess we could suggest a new test phase in cime just for the PFS test that does the same thing as TPUTCOMP but is reported with a different phase name so that it doesn't get excluded when you are ignoring most TPUTCOMP results.)

As you can probably tell, I'm struggling here and don't feel like I have a good answer. Further thoughts are welcome.

Reply:
OK, having this one test sounds acceptable to me. 25% FAIL is too high, maybe 10 or 15%? We could set the threshold for this one test. I suspect machine variability might be around a few percent normally. The problem is that it can also just randomly be higher, any specific time. Maybe we say that if it fails -- you rerun the test a few times to make sure it's consistent? I think in general we are OK with changes in a few percent, but 10% is enough that we start to care.

Reply:
Okay, I have changed the throughput tolerance for the PFS test to 10% (just pushed a commit). Keep in mind that the failure won't show up when you run the standard 'cs.status' script that run_sys_tests creates, as noted above, so you'll need to explicitly check it. @ekluzek if there's any other wording you want to add here, feel free to do so. Once we update cime, we'll be able to say, "grep for TPUTCOMP in the TestStatus.log file in the PFS test" (which will show both the actual tolerance and whether it exceeds the given threshold).